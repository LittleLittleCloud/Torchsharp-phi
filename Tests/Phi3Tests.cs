using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ApprovalTests;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static TorchSharp.torch;
using TorchSharp;
using Xunit;
using FluentAssertions;

namespace Phi.Tests;

public class Phi3Tests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task ShapeTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var model = Phi3ForCasualLM.FromPretrained(modelWeightFolder, torchDtype: ScalarType.BFloat16);
        var state_dict_str = model.Peek();
        Approvals.Verify(state_dict_str);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("Approvals")]
    public async Task TokenizerTest()
    {
        var modelWeightFolder = "C:\\Users\\xiaoyuz\\source\\repos\\Phi-3-mini-4k-instruct";
        var tokenizer = LLama2Tokenizer.FromPretrained(modelWeightFolder);
        tokenizer.BosId.Should().Be(1);
        tokenizer.EosId.Should().Be(32000);
        var messages = new string[]
        {
            "Can you provide ways to eat combinations of bananas and dragonfruits?",
            "Sure! Here are some ways to eat bananas and dragonfruits together: 1. Banana and dragonfruit smoothie: Blend bananas and dragonfruits together with some milk and honey. 2. Banana and dragonfruit salad: Mix sliced bananas and dragonfruits together with some lemon juice and honey.",
            "What about solving an 2x + 3 = 7 equation?"
        };
        var sb = new StringBuilder();
        foreach (var message in messages)
        {
            var tokenized = tokenizer.Encode(message, true, false);
            var tokenized_str = string.Join(", ", tokenized.Select(x => x.ToString()));

            sb.AppendLine(tokenized_str);
        }
        Approvals.Verify(sb.ToString());
    }
}
