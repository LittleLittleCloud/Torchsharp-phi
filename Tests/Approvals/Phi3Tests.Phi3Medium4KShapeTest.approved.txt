0: lm_head.weight: sum: 8805.836  dtype: BFloat16 shape: [32064,5120]
1: model.embed_tokens.weight: sum: 9070.663  dtype: BFloat16 shape: [32064,5120]
2: model.layers.0.input_layernorm.weight: sum: 47.519  dtype: BFloat16 shape: [5120]
3: model.layers.0.mlp.down_proj.weight: sum: 7267.82  dtype: BFloat16 shape: [5120,17920]
4: model.layers.0.mlp.gate_up_proj.weight: sum: 6284.563  dtype: BFloat16 shape: [35840,5120]
5: model.layers.0.post_attention_layernorm.weight: sum: 47.7138  dtype: BFloat16 shape: [5120]
6: model.layers.0.self_attn.o_proj.weight: sum: 3808.5227  dtype: BFloat16 shape: [5120,5120]
7: model.layers.0.self_attn.qkv_proj.weight: sum: 8533.09  dtype: BFloat16 shape: [7680,5120]
8: model.layers.1.input_layernorm.weight: sum: 48.5952  dtype: BFloat16 shape: [5120]
9: model.layers.1.mlp.down_proj.weight: sum: 8560.667  dtype: BFloat16 shape: [5120,17920]
10: model.layers.1.mlp.gate_up_proj.weight: sum: 3721.3516  dtype: BFloat16 shape: [35840,5120]
11: model.layers.1.post_attention_layernorm.weight: sum: 47.6483  dtype: BFloat16 shape: [5120]
12: model.layers.1.self_attn.o_proj.weight: sum: 3603.5671  dtype: BFloat16 shape: [5120,5120]
13: model.layers.1.self_attn.qkv_proj.weight: sum: -1871.2281  dtype: BFloat16 shape: [7680,5120]
14: model.layers.10.input_layernorm.weight: sum: 47.7461  dtype: BFloat16 shape: [5120]
15: model.layers.10.mlp.down_proj.weight: sum: 18187.832  dtype: BFloat16 shape: [5120,17920]
16: model.layers.10.mlp.gate_up_proj.weight: sum: 5630.475  dtype: BFloat16 shape: [35840,5120]
17: model.layers.10.post_attention_layernorm.weight: sum: 47.6267  dtype: BFloat16 shape: [5120]
18: model.layers.10.self_attn.o_proj.weight: sum: -12380.933  dtype: BFloat16 shape: [5120,5120]
19: model.layers.10.self_attn.qkv_proj.weight: sum: 16456.727  dtype: BFloat16 shape: [7680,5120]
20: model.layers.11.input_layernorm.weight: sum: 47.656  dtype: BFloat16 shape: [5120]
21: model.layers.11.mlp.down_proj.weight: sum: 13953.423  dtype: BFloat16 shape: [5120,17920]
22: model.layers.11.mlp.gate_up_proj.weight: sum: 7422.258  dtype: BFloat16 shape: [35840,5120]
23: model.layers.11.post_attention_layernorm.weight: sum: 47.6316  dtype: BFloat16 shape: [5120]
24: model.layers.11.self_attn.o_proj.weight: sum: 3187.4653  dtype: BFloat16 shape: [5120,5120]
25: model.layers.11.self_attn.qkv_proj.weight: sum: 6124.5703  dtype: BFloat16 shape: [7680,5120]
26: model.layers.12.input_layernorm.weight: sum: 47.6027  dtype: BFloat16 shape: [5120]
27: model.layers.12.mlp.down_proj.weight: sum: 4177.265  dtype: BFloat16 shape: [5120,17920]
28: model.layers.12.mlp.gate_up_proj.weight: sum: 10245.885  dtype: BFloat16 shape: [35840,5120]
29: model.layers.12.post_attention_layernorm.weight: sum: 47.6234  dtype: BFloat16 shape: [5120]
30: model.layers.12.self_attn.o_proj.weight: sum: -679.4656  dtype: BFloat16 shape: [5120,5120]
31: model.layers.12.self_attn.qkv_proj.weight: sum: 8041.8594  dtype: BFloat16 shape: [7680,5120]
32: model.layers.13.input_layernorm.weight: sum: 47.6352  dtype: BFloat16 shape: [5120]
33: model.layers.13.mlp.down_proj.weight: sum: 4688.2266  dtype: BFloat16 shape: [5120,17920]
34: model.layers.13.mlp.gate_up_proj.weight: sum: 9860.227  dtype: BFloat16 shape: [35840,5120]
35: model.layers.13.post_attention_layernorm.weight: sum: 47.5705  dtype: BFloat16 shape: [5120]
36: model.layers.13.self_attn.o_proj.weight: sum: 2714.572  dtype: BFloat16 shape: [5120,5120]
37: model.layers.13.self_attn.qkv_proj.weight: sum: 3798.3489  dtype: BFloat16 shape: [7680,5120]
38: model.layers.14.input_layernorm.weight: sum: 47.7116  dtype: BFloat16 shape: [5120]
39: model.layers.14.mlp.down_proj.weight: sum: 2551.9053  dtype: BFloat16 shape: [5120,17920]
40: model.layers.14.mlp.gate_up_proj.weight: sum: 5921.217  dtype: BFloat16 shape: [35840,5120]
41: model.layers.14.post_attention_layernorm.weight: sum: 47.6602  dtype: BFloat16 shape: [5120]
42: model.layers.14.self_attn.o_proj.weight: sum: 2883.7542  dtype: BFloat16 shape: [5120,5120]
43: model.layers.14.self_attn.qkv_proj.weight: sum: 5095.7275  dtype: BFloat16 shape: [7680,5120]
44: model.layers.15.input_layernorm.weight: sum: 47.8162  dtype: BFloat16 shape: [5120]
45: model.layers.15.mlp.down_proj.weight: sum: 8585.163  dtype: BFloat16 shape: [5120,17920]
46: model.layers.15.mlp.gate_up_proj.weight: sum: 227069.56  dtype: BFloat16 shape: [35840,5120]
47: model.layers.15.post_attention_layernorm.weight: sum: 47.6672  dtype: BFloat16 shape: [5120]
48: model.layers.15.self_attn.o_proj.weight: sum: 3766.8547  dtype: BFloat16 shape: [5120,5120]
49: model.layers.15.self_attn.qkv_proj.weight: sum: 5948.155  dtype: BFloat16 shape: [7680,5120]
50: model.layers.16.input_layernorm.weight: sum: 47.6408  dtype: BFloat16 shape: [5120]
51: model.layers.16.mlp.down_proj.weight: sum: 17580.303  dtype: BFloat16 shape: [5120,17920]
52: model.layers.16.mlp.gate_up_proj.weight: sum: -22859.69  dtype: BFloat16 shape: [35840,5120]
53: model.layers.16.post_attention_layernorm.weight: sum: 47.7359  dtype: BFloat16 shape: [5120]
54: model.layers.16.self_attn.o_proj.weight: sum: 2923.803  dtype: BFloat16 shape: [5120,5120]
55: model.layers.16.self_attn.qkv_proj.weight: sum: 6114.5225  dtype: BFloat16 shape: [7680,5120]
56: model.layers.17.input_layernorm.weight: sum: 47.6887  dtype: BFloat16 shape: [5120]
57: model.layers.17.mlp.down_proj.weight: sum: 19817.22  dtype: BFloat16 shape: [5120,17920]
58: model.layers.17.mlp.gate_up_proj.weight: sum: -12457.958  dtype: BFloat16 shape: [35840,5120]
59: model.layers.17.post_attention_layernorm.weight: sum: 47.5615  dtype: BFloat16 shape: [5120]
60: model.layers.17.self_attn.o_proj.weight: sum: 2758.6843  dtype: BFloat16 shape: [5120,5120]
61: model.layers.17.self_attn.qkv_proj.weight: sum: 5464.8315  dtype: BFloat16 shape: [7680,5120]
62: model.layers.18.input_layernorm.weight: sum: 47.6541  dtype: BFloat16 shape: [5120]
63: model.layers.18.mlp.down_proj.weight: sum: 5466.238  dtype: BFloat16 shape: [5120,17920]
64: model.layers.18.mlp.gate_up_proj.weight: sum: 4616.182  dtype: BFloat16 shape: [35840,5120]
65: model.layers.18.post_attention_layernorm.weight: sum: 47.794  dtype: BFloat16 shape: [5120]
66: model.layers.18.self_attn.o_proj.weight: sum: 4686.807  dtype: BFloat16 shape: [5120,5120]
67: model.layers.18.self_attn.qkv_proj.weight: sum: 689.7584  dtype: BFloat16 shape: [7680,5120]
68: model.layers.19.input_layernorm.weight: sum: 47.7848  dtype: BFloat16 shape: [5120]
69: model.layers.19.mlp.down_proj.weight: sum: 5775.1616  dtype: BFloat16 shape: [5120,17920]
70: model.layers.19.mlp.gate_up_proj.weight: sum: 5488.2056  dtype: BFloat16 shape: [35840,5120]
71: model.layers.19.post_attention_layernorm.weight: sum: 47.7105  dtype: BFloat16 shape: [5120]
72: model.layers.19.self_attn.o_proj.weight: sum: 4137.018  dtype: BFloat16 shape: [5120,5120]
73: model.layers.19.self_attn.qkv_proj.weight: sum: 4108.4805  dtype: BFloat16 shape: [7680,5120]
74: model.layers.2.input_layernorm.weight: sum: 47.8654  dtype: BFloat16 shape: [5120]
75: model.layers.2.mlp.down_proj.weight: sum: -10707.934  dtype: BFloat16 shape: [5120,17920]
76: model.layers.2.mlp.gate_up_proj.weight: sum: 6479.9385  dtype: BFloat16 shape: [35840,5120]
77: model.layers.2.post_attention_layernorm.weight: sum: 47.5718  dtype: BFloat16 shape: [5120]
78: model.layers.2.self_attn.o_proj.weight: sum: 5149.244  dtype: BFloat16 shape: [5120,5120]
79: model.layers.2.self_attn.qkv_proj.weight: sum: 4962.083  dtype: BFloat16 shape: [7680,5120]
80: model.layers.20.input_layernorm.weight: sum: 47.649  dtype: BFloat16 shape: [5120]
81: model.layers.20.mlp.down_proj.weight: sum: 7209.5913  dtype: BFloat16 shape: [5120,17920]
82: model.layers.20.mlp.gate_up_proj.weight: sum: 7091.6455  dtype: BFloat16 shape: [35840,5120]
83: model.layers.20.post_attention_layernorm.weight: sum: 47.8034  dtype: BFloat16 shape: [5120]
84: model.layers.20.self_attn.o_proj.weight: sum: 6437.262  dtype: BFloat16 shape: [5120,5120]
85: model.layers.20.self_attn.qkv_proj.weight: sum: -1000.0214  dtype: BFloat16 shape: [7680,5120]
86: model.layers.21.input_layernorm.weight: sum: 47.6604  dtype: BFloat16 shape: [5120]
87: model.layers.21.mlp.down_proj.weight: sum: 9937.241  dtype: BFloat16 shape: [5120,17920]
88: model.layers.21.mlp.gate_up_proj.weight: sum: 8298.995  dtype: BFloat16 shape: [35840,5120]
89: model.layers.21.post_attention_layernorm.weight: sum: 47.6229  dtype: BFloat16 shape: [5120]
90: model.layers.21.self_attn.o_proj.weight: sum: 1840.2474  dtype: BFloat16 shape: [5120,5120]
91: model.layers.21.self_attn.qkv_proj.weight: sum: 1287.8334  dtype: BFloat16 shape: [7680,5120]
92: model.layers.22.input_layernorm.weight: sum: 47.5764  dtype: BFloat16 shape: [5120]
93: model.layers.22.mlp.down_proj.weight: sum: 5584.5005  dtype: BFloat16 shape: [5120,17920]
94: model.layers.22.mlp.gate_up_proj.weight: sum: 6010.6543  dtype: BFloat16 shape: [35840,5120]
95: model.layers.22.post_attention_layernorm.weight: sum: 47.6253  dtype: BFloat16 shape: [5120]
96: model.layers.22.self_attn.o_proj.weight: sum: 3414.1064  dtype: BFloat16 shape: [5120,5120]
97: model.layers.22.self_attn.qkv_proj.weight: sum: 6152.8096  dtype: BFloat16 shape: [7680,5120]
98: model.layers.23.input_layernorm.weight: sum: 47.8164  dtype: BFloat16 shape: [5120]
99: model.layers.23.mlp.down_proj.weight: sum: 6413.343  dtype: BFloat16 shape: [5120,17920]
100: model.layers.23.mlp.gate_up_proj.weight: sum: 16406.033  dtype: BFloat16 shape: [35840,5120]
101: model.layers.23.post_attention_layernorm.weight: sum: 47.7422  dtype: BFloat16 shape: [5120]
102: model.layers.23.self_attn.o_proj.weight: sum: 4741.392  dtype: BFloat16 shape: [5120,5120]
103: model.layers.23.self_attn.qkv_proj.weight: sum: 3239.9102  dtype: BFloat16 shape: [7680,5120]
104: model.layers.24.input_layernorm.weight: sum: 47.7046  dtype: BFloat16 shape: [5120]
105: model.layers.24.mlp.down_proj.weight: sum: 8112.3022  dtype: BFloat16 shape: [5120,17920]
106: model.layers.24.mlp.gate_up_proj.weight: sum: 6524.558  dtype: BFloat16 shape: [35840,5120]
107: model.layers.24.post_attention_layernorm.weight: sum: 47.7843  dtype: BFloat16 shape: [5120]
108: model.layers.24.self_attn.o_proj.weight: sum: 2643.2905  dtype: BFloat16 shape: [5120,5120]
109: model.layers.24.self_attn.qkv_proj.weight: sum: 5582.107  dtype: BFloat16 shape: [7680,5120]
110: model.layers.25.input_layernorm.weight: sum: 47.6526  dtype: BFloat16 shape: [5120]
111: model.layers.25.mlp.down_proj.weight: sum: 9264.35  dtype: BFloat16 shape: [5120,17920]
112: model.layers.25.mlp.gate_up_proj.weight: sum: 7591.327  dtype: BFloat16 shape: [35840,5120]
113: model.layers.25.post_attention_layernorm.weight: sum: 47.703  dtype: BFloat16 shape: [5120]
114: model.layers.25.self_attn.o_proj.weight: sum: 7787.1665  dtype: BFloat16 shape: [5120,5120]
115: model.layers.25.self_attn.qkv_proj.weight: sum: 7905.8794  dtype: BFloat16 shape: [7680,5120]
116: model.layers.26.input_layernorm.weight: sum: 47.707  dtype: BFloat16 shape: [5120]
117: model.layers.26.mlp.down_proj.weight: sum: 7953.0586  dtype: BFloat16 shape: [5120,17920]
118: model.layers.26.mlp.gate_up_proj.weight: sum: 6639.5737  dtype: BFloat16 shape: [35840,5120]
119: model.layers.26.post_attention_layernorm.weight: sum: 47.5223  dtype: BFloat16 shape: [5120]
120: model.layers.26.self_attn.o_proj.weight: sum: 4998.194  dtype: BFloat16 shape: [5120,5120]
121: model.layers.26.self_attn.qkv_proj.weight: sum: 3809.7288  dtype: BFloat16 shape: [7680,5120]
122: model.layers.27.input_layernorm.weight: sum: 47.7693  dtype: BFloat16 shape: [5120]
123: model.layers.27.mlp.down_proj.weight: sum: 23033.664  dtype: BFloat16 shape: [5120,17920]
124: model.layers.27.mlp.gate_up_proj.weight: sum: 15441.764  dtype: BFloat16 shape: [35840,5120]
125: model.layers.27.post_attention_layernorm.weight: sum: 47.6885  dtype: BFloat16 shape: [5120]
126: model.layers.27.self_attn.o_proj.weight: sum: 2555.9363  dtype: BFloat16 shape: [5120,5120]
127: model.layers.27.self_attn.qkv_proj.weight: sum: 3794.089  dtype: BFloat16 shape: [7680,5120]
128: model.layers.28.input_layernorm.weight: sum: 47.6447  dtype: BFloat16 shape: [5120]
129: model.layers.28.mlp.down_proj.weight: sum: 7603.5625  dtype: BFloat16 shape: [5120,17920]
130: model.layers.28.mlp.gate_up_proj.weight: sum: 7764.163  dtype: BFloat16 shape: [35840,5120]
131: model.layers.28.post_attention_layernorm.weight: sum: 47.7393  dtype: BFloat16 shape: [5120]
132: model.layers.28.self_attn.o_proj.weight: sum: 1784.6428  dtype: BFloat16 shape: [5120,5120]
133: model.layers.28.self_attn.qkv_proj.weight: sum: 3744.6367  dtype: BFloat16 shape: [7680,5120]
134: model.layers.29.input_layernorm.weight: sum: 47.6657  dtype: BFloat16 shape: [5120]
135: model.layers.29.mlp.down_proj.weight: sum: 50437.01  dtype: BFloat16 shape: [5120,17920]
136: model.layers.29.mlp.gate_up_proj.weight: sum: 6860.2095  dtype: BFloat16 shape: [35840,5120]
137: model.layers.29.post_attention_layernorm.weight: sum: 47.5428  dtype: BFloat16 shape: [5120]
138: model.layers.29.self_attn.o_proj.weight: sum: 3384.5503  dtype: BFloat16 shape: [5120,5120]
139: model.layers.29.self_attn.qkv_proj.weight: sum: 9099.484  dtype: BFloat16 shape: [7680,5120]
140: model.layers.3.input_layernorm.weight: sum: 47.7664  dtype: BFloat16 shape: [5120]
141: model.layers.3.mlp.down_proj.weight: sum: 6695.718  dtype: BFloat16 shape: [5120,17920]
142: model.layers.3.mlp.gate_up_proj.weight: sum: 7311.063  dtype: BFloat16 shape: [35840,5120]
143: model.layers.3.post_attention_layernorm.weight: sum: 47.6748  dtype: BFloat16 shape: [5120]
144: model.layers.3.self_attn.o_proj.weight: sum: 3621.8855  dtype: BFloat16 shape: [5120,5120]
145: model.layers.3.self_attn.qkv_proj.weight: sum: 2753.5146  dtype: BFloat16 shape: [7680,5120]
146: model.layers.30.input_layernorm.weight: sum: 47.8046  dtype: BFloat16 shape: [5120]
147: model.layers.30.mlp.down_proj.weight: sum: 11056.54  dtype: BFloat16 shape: [5120,17920]
148: model.layers.30.mlp.gate_up_proj.weight: sum: 5278.9478  dtype: BFloat16 shape: [35840,5120]
149: model.layers.30.post_attention_layernorm.weight: sum: 47.6428  dtype: BFloat16 shape: [5120]
150: model.layers.30.self_attn.o_proj.weight: sum: 2646.2048  dtype: BFloat16 shape: [5120,5120]
151: model.layers.30.self_attn.qkv_proj.weight: sum: -80775.62  dtype: BFloat16 shape: [7680,5120]
152: model.layers.31.input_layernorm.weight: sum: 47.7874  dtype: BFloat16 shape: [5120]
153: model.layers.31.mlp.down_proj.weight: sum: 15318.682  dtype: BFloat16 shape: [5120,17920]
154: model.layers.31.mlp.gate_up_proj.weight: sum: 11929.647  dtype: BFloat16 shape: [35840,5120]
155: model.layers.31.post_attention_layernorm.weight: sum: 47.7614  dtype: BFloat16 shape: [5120]
156: model.layers.31.self_attn.o_proj.weight: sum: 4183.699  dtype: BFloat16 shape: [5120,5120]
157: model.layers.31.self_attn.qkv_proj.weight: sum: 6571.714  dtype: BFloat16 shape: [7680,5120]
158: model.layers.32.input_layernorm.weight: sum: 47.5937  dtype: BFloat16 shape: [5120]
159: model.layers.32.mlp.down_proj.weight: sum: 3109.7747  dtype: BFloat16 shape: [5120,17920]
160: model.layers.32.mlp.gate_up_proj.weight: sum: 13138.158  dtype: BFloat16 shape: [35840,5120]
161: model.layers.32.post_attention_layernorm.weight: sum: 47.6423  dtype: BFloat16 shape: [5120]
162: model.layers.32.self_attn.o_proj.weight: sum: 2371.7473  dtype: BFloat16 shape: [5120,5120]
163: model.layers.32.self_attn.qkv_proj.weight: sum: 5218.907  dtype: BFloat16 shape: [7680,5120]
164: model.layers.33.input_layernorm.weight: sum: 47.6873  dtype: BFloat16 shape: [5120]
165: model.layers.33.mlp.down_proj.weight: sum: -171379.48  dtype: BFloat16 shape: [5120,17920]
166: model.layers.33.mlp.gate_up_proj.weight: sum: 10236.684  dtype: BFloat16 shape: [35840,5120]
167: model.layers.33.post_attention_layernorm.weight: sum: 47.789  dtype: BFloat16 shape: [5120]
168: model.layers.33.self_attn.o_proj.weight: sum: 2572.0183  dtype: BFloat16 shape: [5120,5120]
169: model.layers.33.self_attn.qkv_proj.weight: sum: 4193.863  dtype: BFloat16 shape: [7680,5120]
170: model.layers.34.input_layernorm.weight: sum: 47.7731  dtype: BFloat16 shape: [5120]
171: model.layers.34.mlp.down_proj.weight: sum: 7606.99  dtype: BFloat16 shape: [5120,17920]
172: model.layers.34.mlp.gate_up_proj.weight: sum: 6819.317  dtype: BFloat16 shape: [35840,5120]
173: model.layers.34.post_attention_layernorm.weight: sum: 47.5843  dtype: BFloat16 shape: [5120]
174: model.layers.34.self_attn.o_proj.weight: sum: 3181.2517  dtype: BFloat16 shape: [5120,5120]
175: model.layers.34.self_attn.qkv_proj.weight: sum: -85.9981  dtype: BFloat16 shape: [7680,5120]
176: model.layers.35.input_layernorm.weight: sum: 47.687  dtype: BFloat16 shape: [5120]
177: model.layers.35.mlp.down_proj.weight: sum: 6781.886  dtype: BFloat16 shape: [5120,17920]
178: model.layers.35.mlp.gate_up_proj.weight: sum: 4906.2744  dtype: BFloat16 shape: [35840,5120]
179: model.layers.35.post_attention_layernorm.weight: sum: 47.7155  dtype: BFloat16 shape: [5120]
180: model.layers.35.self_attn.o_proj.weight: sum: 771.6975  dtype: BFloat16 shape: [5120,5120]
181: model.layers.35.self_attn.qkv_proj.weight: sum: 11902.68  dtype: BFloat16 shape: [7680,5120]
182: model.layers.36.input_layernorm.weight: sum: 47.6995  dtype: BFloat16 shape: [5120]
183: model.layers.36.mlp.down_proj.weight: sum: 4798.4424  dtype: BFloat16 shape: [5120,17920]
184: model.layers.36.mlp.gate_up_proj.weight: sum: -126.6086  dtype: BFloat16 shape: [35840,5120]
185: model.layers.36.post_attention_layernorm.weight: sum: 47.6454  dtype: BFloat16 shape: [5120]
186: model.layers.36.self_attn.o_proj.weight: sum: 1574.9421  dtype: BFloat16 shape: [5120,5120]
187: model.layers.36.self_attn.qkv_proj.weight: sum: 5262.006  dtype: BFloat16 shape: [7680,5120]
188: model.layers.37.input_layernorm.weight: sum: 47.6963  dtype: BFloat16 shape: [5120]
189: model.layers.37.mlp.down_proj.weight: sum: 8800.9795  dtype: BFloat16 shape: [5120,17920]
190: model.layers.37.mlp.gate_up_proj.weight: sum: 6528.2466  dtype: BFloat16 shape: [35840,5120]
191: model.layers.37.post_attention_layernorm.weight: sum: 47.6788  dtype: BFloat16 shape: [5120]
192: model.layers.37.self_attn.o_proj.weight: sum: 766.438  dtype: BFloat16 shape: [5120,5120]
193: model.layers.37.self_attn.qkv_proj.weight: sum: -1349.1729  dtype: BFloat16 shape: [7680,5120]
194: model.layers.38.input_layernorm.weight: sum: 47.6972  dtype: BFloat16 shape: [5120]
195: model.layers.38.mlp.down_proj.weight: sum: 2601.455  dtype: BFloat16 shape: [5120,17920]
196: model.layers.38.mlp.gate_up_proj.weight: sum: -5983.464  dtype: BFloat16 shape: [35840,5120]
197: model.layers.38.post_attention_layernorm.weight: sum: 47.6438  dtype: BFloat16 shape: [5120]
198: model.layers.38.self_attn.o_proj.weight: sum: 3588.78  dtype: BFloat16 shape: [5120,5120]
199: model.layers.38.self_attn.qkv_proj.weight: sum: 4928.921  dtype: BFloat16 shape: [7680,5120]
200: model.layers.39.input_layernorm.weight: sum: 47.6634  dtype: BFloat16 shape: [5120]
201: model.layers.39.mlp.down_proj.weight: sum: 7309.1426  dtype: BFloat16 shape: [5120,17920]
202: model.layers.39.mlp.gate_up_proj.weight: sum: -9873.352  dtype: BFloat16 shape: [35840,5120]
203: model.layers.39.post_attention_layernorm.weight: sum: 47.7071  dtype: BFloat16 shape: [5120]
204: model.layers.39.self_attn.o_proj.weight: sum: 2984.3765  dtype: BFloat16 shape: [5120,5120]
205: model.layers.39.self_attn.qkv_proj.weight: sum: 5339.7373  dtype: BFloat16 shape: [7680,5120]
206: model.layers.4.input_layernorm.weight: sum: 47.696  dtype: BFloat16 shape: [5120]
207: model.layers.4.mlp.down_proj.weight: sum: -6401.727  dtype: BFloat16 shape: [5120,17920]
208: model.layers.4.mlp.gate_up_proj.weight: sum: 6110.9565  dtype: BFloat16 shape: [35840,5120]
209: model.layers.4.post_attention_layernorm.weight: sum: 47.6948  dtype: BFloat16 shape: [5120]
210: model.layers.4.self_attn.o_proj.weight: sum: 3143.1045  dtype: BFloat16 shape: [5120,5120]
211: model.layers.4.self_attn.qkv_proj.weight: sum: 5102.626  dtype: BFloat16 shape: [7680,5120]
212: model.layers.5.input_layernorm.weight: sum: 47.626  dtype: BFloat16 shape: [5120]
213: model.layers.5.mlp.down_proj.weight: sum: 6340.7466  dtype: BFloat16 shape: [5120,17920]
214: model.layers.5.mlp.gate_up_proj.weight: sum: 5601.7656  dtype: BFloat16 shape: [35840,5120]
215: model.layers.5.post_attention_layernorm.weight: sum: 47.8421  dtype: BFloat16 shape: [5120]
216: model.layers.5.self_attn.o_proj.weight: sum: 4348.728  dtype: BFloat16 shape: [5120,5120]
217: model.layers.5.self_attn.qkv_proj.weight: sum: 2783.7664  dtype: BFloat16 shape: [7680,5120]
218: model.layers.6.input_layernorm.weight: sum: 47.8099  dtype: BFloat16 shape: [5120]
219: model.layers.6.mlp.down_proj.weight: sum: 6948.468  dtype: BFloat16 shape: [5120,17920]
220: model.layers.6.mlp.gate_up_proj.weight: sum: 6209.047  dtype: BFloat16 shape: [35840,5120]
221: model.layers.6.post_attention_layernorm.weight: sum: 47.6644  dtype: BFloat16 shape: [5120]
222: model.layers.6.self_attn.o_proj.weight: sum: 2609.1765  dtype: BFloat16 shape: [5120,5120]
223: model.layers.6.self_attn.qkv_proj.weight: sum: 11593.47  dtype: BFloat16 shape: [7680,5120]
224: model.layers.7.input_layernorm.weight: sum: 47.6772  dtype: BFloat16 shape: [5120]
225: model.layers.7.mlp.down_proj.weight: sum: 2708.6248  dtype: BFloat16 shape: [5120,17920]
226: model.layers.7.mlp.gate_up_proj.weight: sum: 4598.566  dtype: BFloat16 shape: [35840,5120]
227: model.layers.7.post_attention_layernorm.weight: sum: 47.5788  dtype: BFloat16 shape: [5120]
228: model.layers.7.self_attn.o_proj.weight: sum: 3314.7402  dtype: BFloat16 shape: [5120,5120]
229: model.layers.7.self_attn.qkv_proj.weight: sum: 5000.466  dtype: BFloat16 shape: [7680,5120]
230: model.layers.8.input_layernorm.weight: sum: 47.5573  dtype: BFloat16 shape: [5120]
231: model.layers.8.mlp.down_proj.weight: sum: -13125.318  dtype: BFloat16 shape: [5120,17920]
232: model.layers.8.mlp.gate_up_proj.weight: sum: 2802.7253  dtype: BFloat16 shape: [35840,5120]
233: model.layers.8.post_attention_layernorm.weight: sum: 47.7284  dtype: BFloat16 shape: [5120]
234: model.layers.8.self_attn.o_proj.weight: sum: 1383.8076  dtype: BFloat16 shape: [5120,5120]
235: model.layers.8.self_attn.qkv_proj.weight: sum: 5332.6167  dtype: BFloat16 shape: [7680,5120]
236: model.layers.9.input_layernorm.weight: sum: 47.5792  dtype: BFloat16 shape: [5120]
237: model.layers.9.mlp.down_proj.weight: sum: 5747.059  dtype: BFloat16 shape: [5120,17920]
238: model.layers.9.mlp.gate_up_proj.weight: sum: 6835.9287  dtype: BFloat16 shape: [35840,5120]
239: model.layers.9.post_attention_layernorm.weight: sum: 47.5875  dtype: BFloat16 shape: [5120]
240: model.layers.9.self_attn.o_proj.weight: sum: 5110.1963  dtype: BFloat16 shape: [5120,5120]
241: model.layers.9.self_attn.qkv_proj.weight: sum: -4071.3303  dtype: BFloat16 shape: [7680,5120]
242: model.norm.weight: sum: 47.6743  dtype: BFloat16 shape: [5120]
